contracts/verify/Verifier.sol

```diff
@@ -63,7 +63,7 @@ contract Verifier is Ownable {
       }
   }
 
-  function verificationCalculation(uint256[] memory _proof, uint256[] memory _publicInputs, uint256[] memory _vk) public returns (uint) {
+  function verificationCalculation(uint256[] calldata _proof, uint256[] calldata _publicInputs, uint256[] calldata _vk) public returns (uint) {
```

Comment:
Overall this is great - however I expected the tests to fail here since all instances of `_vk` originally come from storage in the shield contract. For that reason I think they can only be converted to `memory` at best - however `_proof` may work fine